src/registry/LocalRegistry.ts

```diff
@@ -0,0 +1,101 @@
+import fs from 'fs';
+import path from 'path';
+import type { Logger } from 'pino';
+import { parse as yamlParse } from 'yaml';
+
+import {
+  ChainMetadataSchema,
+  type ChainMap,
+  type ChainMetadata,
+  type ChainName,
+} from '@hyperlane-xyz/sdk';
+
+import { ChainAddresses, ChainAddressesSchema } from '../types.js';
+import { BaseRegistry, CHAIN_FILE_REGEX } from './BaseRegistry.js';
+import {
+  RegistryType,
+  type ChainFiles,
+  type IRegistry,
+  type RegistryContent,
+} from './IRegistry.js';
+
+export interface LocalRegistryOptions {
+  uri: string;
+  logger?: Logger;
+}
+
+export class LocalRegistry extends BaseRegistry implements IRegistry {
+  public readonly type = RegistryType.Local;
+  public readonly uri: string;
+
+  constructor(options: LocalRegistryOptions) {
+    super({ logger: options.logger });
+    this.uri = options.uri;
+  }
+
+  listRegistryContent(): RegistryContent {
+    if (this.listContentCache) return this.listContentCache;
+
+    const chainFileList = this.listFiles(path.join(this.uri, this.getChainsPath()));
+    const chains: ChainMap<ChainFiles> = {};
```

Comment:
There's some code duplication between this file and GithubRegistry but importantly, all the methods here are sync, which makes code reuse tricky. Given that it's <30 dupe lines total, I'm just going with this for now. We can revisit if more duplication arises when adding the write-related methods

Reply:
could you use your MaybePromise thing somehow?

Reply:
having all of the per/multi chain stuff be reused might be nice